### ISSUE TITLE
Python 2.7 removed from macOS 12.3 = broken workflow

### ISSUE THREAD

**Issue by rhydlewis** [rhydlewis/search-omnifocus#50]
- Reported in [early March](https://twitter.com/OliverJHaslam/status/1498994323253248002)
- Workaround is to reinstall Python until a new release is available

**Comment by Doslin** [comment]
Will you rewrite this workflow

> * Reported in [early March](https://twitter.com/OliverJHaslam/status/1498994323253248002)
> * Workaround is to reinstall Python until a new release is available

**Comment by rhydlewis** [comment]
Yes, I have a basic version of the workflow using javascript in progress at the moment. Aiming to release it for evaluation next week

**Comment by rhydlewis** [comment]
Temporary workaround 👉 [https://www.alfredapp.com/help/kb/python-2-monterey/](https://www.alfredapp.com/help/kb/python-2-monterey/)

**Comment by Doslin** [comment]
> Temporary workaround 👉 https://www.alfredapp.com/help/kb/python-2-monterey/
It didn't work

Last 10 log lines:
ld: warning: directory not found for option '-L/Users/zhilin/.pyenv/versions/2.7.18/lib'
ld: warning: ignoring file libpython2.7.a, building for macOS-x86_64 but attempting to link with file built for unknown-unsupported file format ( 0x21 0x3C 0x61 0x72 0x63 0x68 0x3E 0x0A 0x2F 0x20 0x20 0x20 0x20 0x20 0x20 0x20 )
Undefined symbols for architecture x86_64:
  "_PyMac_Error", referenced from:
     -u command line option
  "_Py_Main", referenced from:
      _main in python.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [python.exe] Error 1

**Comment by Doslin** [comment]
> @Doslin Not sure the installation of Python2 has worked correctly for you? The workaround from Alfred is working fine for me.
Meanwhile, can I help you with this version of javascript? I'm looking forward to this version very much

**Comment by Doslin** [comment]
Can't find the python
```shell
ln -s "~/.pyenv/versions/2.7.18/bin/python2.7" "/usr/bin/python"
```
ln: /usr/bin/python: Read-only file system
``` log

[13:37:41.238] Logging Stopped.
[13:37:42.084] Logging Started...
[13:37:47.843] Search OmniFocus[[Script Filter](alfredpreferences:workflows%3Eworkflow%3Euser.workflow.E20D9C1B-4305-4FA2-9A65-7B65C28898C2%3EF8E62EE2-D96A-4A26-8E39-EBBEFCE2DF97)] Queuing argument 's'
[13:37:47.912] Search OmniFocus[[Script Filter](alfredpreferences:workflows%3Eworkflow%3Euser.workflow.E20D9C1B-4305-4FA2-9A65-7B65C28898C2%3EF8E62EE2-D96A-4A26-8E39-EBBEFCE2DF97)] Script with argv '(null)' finished
[13:37:47.914] ERROR: Search OmniFocus[[Script Filter](alfredpreferences:workflows%3Eworkflow%3Euser.workflow.E20D9C1B-4305-4FA2-9A65-7B65C28898C2%3EF8E62EE2-D96A-4A26-8E39-EBBEFCE2DF97)] Code 127: /bin/bash: /usr/bin/python: No such file or directory
```

**Comment by neuromancer3142** [comment]
Dear rhydlewis,

Unfortunately, the 

> Temporary workaround 👉 https://www.alfredapp.com/help/kb/python-2-monterey/

did not work for me. Homebrew installed everything correctly, but I cannot launch the shortcuts using Alfred. Do you have any tips on how to troubleshoot this? No idea where to start.

Alternative would be to directly use the new javascript version. Do you know when you will release this update? I use your workflow many times per day and dearly miss this great functionality. Thanks for letting me know.

Best regards

**Comment by Doslin** [comment]
https://github.com/alfredapp/updated-third-party-python2-workflows
## Updated Third-Party Python 2 Workflows
With macOS Monterey 12.3, Apple [removed the bundled Python 2 interpreter](https://developer.apple.com/documentation/macos-release-notes/macos-12_3-release-notes#Python). That means older Workflows dependent on Python 2 will no longer work without modification.

That’s what this repository is for. It contains existing third-party Python 2 Workflows tweaked so you can keep using them until they are updated by their developers or you switch to an alternative. If a Workflow you use isn’t present, request it [here](https://github.com/alfredapp/updated-third-party-python2-workflows/issues/new?template=request_workflow.yml) or [on the forum](https://www.alfredforum.com/topic/17941-making-python-2-workflows-work-on-macos-monterey-123-and-above/).

Before installing any of these, you must follow [the knowledge base article](https://www.alfredapp.com/help/kb/python-2-monterey/) to install Python 2 to a location recognised by Alfred.

> WARNING: All Workflows in this repository are made by third-parties. They were modified just enough to keep working but their contents have not been audited so make sure you trust the original author.

> All Workflows retain their original license.

**Comment by rhydlewis** [comment]
Replacement workflow but still WIP 👉 [https://github.com/rhydlewis/alfred-search-omnifocus](https://github.com/rhydlewis/alfred-search-omnifocus)

**Comment by neuromancer3142** [comment]
> https://github.com/alfredapp/updated-third-party-python2-workflows
> ## Updated Third-Party Python 2 Workflows
> 
> With macOS Monterey 12.3, Apple [removed the bundled Python 2 interpreter](https://developer.apple.com/documentation/macos-release-notes/macos-12_3-release-notes#Python). That means older Workflows dependent on Python 2 will no longer work without modification.
> 
> That’s what this repository is for. It contains existing third-party Python 2 Workflows tweaked so you can keep using them until they are updated by their developers or you switch to an alternative. If a Workflow you use isn’t present, request it [here](https://github.com/alfredapp/updated-third-party-python2-workflows/issues/new?template=request_workflow.yml) or [on the forum](https://www.alfredforum.com/topic/17941-making-python-2-workflows-work-on-macos-monterey-123-and-above/).
> 
> Before installing any of these, you must follow [the knowledge base article](https://www.alfredapp.com/help/kb/python-2-monterey/) to install Python 2 to a location recognised by Alfred.
> 
> > WARNING: All Workflows in this repository are made by third-parties. They were modified just enough to keep working but their contents have not been audited so make sure you trust the original author.
> 
> > All Workflows retain their original license.

Dear Doslin,

Finally I got it to work again. Thanks so much. I previously missed the link to the updated script.

Best regards
